subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -665,4 +634,363 @@ static struct lll_df_adv_cfg *df_adv_cfg_acquire(void)
+			 * transmission.
+			 */
+			return err;
+		}
+		pdu_prev = pdu_next;
+	}
+
+	/* If there is missing only one CTE do not add aux_ptr to PDU */
+	if (cte_count - cte_index >= 2) {
+		pdu_add_field_flags |= ULL_ADV_PDU_HDR_FIELD_AUX_PTR;
+	} else {
+		pdu_add_field_flags = ULL_ADV_PDU_HDR_FIELD_CTE_INFO;
+	}
+
+	/* Add new PDUs if the number of PDUs in existing chain is lower than requested number
+	 * of CTEs.
+	 */
+	while (cte_index < cte_count) {
+		pdu_prev = pdu;
+		pdu = lll_adv_pdu_alloc_pdu_adv();
+		if (!pdu) {
+			/* TODO: implement graceful error handling, cleanup of
+			 * changed PDUs.
+			 */
+			return BT_HCI_ERR_MEM_CAPACITY_EXCEEDED;
+		}
+		ull_adv_sync_pdu_init(pdu, pdu_add_field_flags);
+		ull_adv_sync_pdu_cte_info_set(pdu, cte_info);
+
+		/* Link PDU into a chain */
+		lll_adv_pdu_linked_append(pdu, pdu_prev);
+
+		++cte_index;
+		/* If next PDU in a chain is last PDU, then remove aux_ptr field flag from
+		 * extended advertising header.
+		 */
+		if (cte_index == cte_count - 1) {
+			pdu_add_field_flags &= (~ULL_ADV_PDU_HDR_FIELD_AUX_PTR);
+		}
+	}
+
+	return BT_HCI_ERR_SUCCESS;
+}
+#endif /* CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX > 1 */
+
+/*
+ * @brief Function sets content of cte_info field for periodic advertising
+ *
+ * @param adv            Pointer to periodic advertising set.
+ * @param df_cfg         Pointer to direction finding configuration
+ * @param[out] ter_idx   Pointer used to return index of allocated or updated PDU.
+ *                       Index is required for scheduling the PDU for transmission in LLL.
+ * @param[out] first_pdu Pointer to return address of first PDU in a periodic advertising chain
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+static uint8_t set_cte_info(struct ll_adv_set *adv, struct lll_df_adv_cfg *df_cfg, uint8_t *ter_idx,
+			    struct pdu_adv **first_pdu)
+{
+	struct pdu_adv *pdu_prev, *pdu;
+	struct lll_adv_sync *lll_sync;
+	struct pdu_cte_info cte_info;
+	uint8_t pdu_add_field_flags;
+	void *extra_data;
+	uint8_t err;
+
+	lll_sync = adv->lll.sync;
+
+	cte_info.type = df_cfg->cte_type;
+	cte_info.time = df_cfg->cte_length;
+
+	/* Note: ULL_ADV_PDU_HDR_FIELD_CTE_INFO is just information that extra_data
+	 * is required in case of this ull_adv_sync_pdu_alloc.
+	 * Other flags here do not change anything. It may be changed to use
+	 * true/false flag for extra_data allocation.
+	 */
+	err = ull_adv_sync_pdu_alloc(adv, ULL_ADV_PDU_EXTRA_DATA_ALLOC_ALWAYS, &pdu_prev, &pdu,
+				     NULL, &extra_data, ter_idx);
+	if (err != BT_HCI_ERR_SUCCESS) {
+		return err;
+	}
+
+	ull_adv_sync_extra_data_set_clear(NULL, extra_data, ULL_ADV_PDU_HDR_FIELD_CTE_INFO, 0,
+					  df_cfg);
+
+#if (CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX > 1)
+	if (df_cfg->cte_count > 1) {
+		pdu_add_field_flags =
+			ULL_ADV_PDU_HDR_FIELD_CTE_INFO | ULL_ADV_PDU_HDR_FIELD_AUX_PTR;
+	} else
+#endif /* CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX > 1 */
```

Comment:
```suggestion
	if (CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX > 1 && 
	    df_cfg->cte_count > 1) {
		pdu_add_field_flags =
			ULL_ADV_PDU_HDR_FIELD_CTE_INFO | ULL_ADV_PDU_HDR_FIELD_AUX_PTR;
	} else
```

Reply:
This is not possible. `BT_CTLR_DF_PER_ADV_CTE_NUM_MAX`  dependends on `BT_CTLR_ADV_PDU_LINK`. If one wanted to use single CTE and didn't enable chaining PUDs that code would not build.